scrapy/utils/python.py

```diff
@@ -198,10 +197,30 @@ def binary_is_text(data):
     return all(c not in _BINARYCHARS for c in data)
 
 
+def getargspec_py23(func):
+    """getargspec_py23(function) -> named tuple ArgSpec(args, varargs, keywords,
+                                                        defaults)
+
+    Identical to inspect.getargspec() in python2, but uses
+    inspect.getfullargspec() for python3 behind the scenes to avoid
+    DeprecationWarning.
+
+    >>> def f(a, b=2, *ar, **kw):
+    ...     pass
+
+    >>> getargspec_py23(f)
+    ArgSpec(args=['a', 'b'], varargs='ar', keywords='kw', defaults=(2,))
+    """
+    if six.PY3:
```

Comment:
And I would make this function private, so we reserve the right to drop it if not needed by other utils in this module.